inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/ConceptFeatureEditor.java

```diff
@@ -141,71 +142,146 @@ private JCas getEditorCas(AnnotationActionHandler aHandler) throws IOException
+                .find(IPartialPageRequestHandler.class);
+            if (target != null) {
+                target.addChildren(getPage(), IFeedback.class);
+            }
+        }
+        return handles;
+    }
 
-            if (traits.getRepositoryId() != null) {
-                // If a specific KB is selected, get its instances
-                Optional<KnowledgeBase> kb = kbService.getKnowledgeBaseById(project,
-                        traits.getRepositoryId());
-                if (kb.isPresent()) {
-                    if (kb.get().isSupportConceptLinking()) {
-                        handles.addAll(listLinkingInstances(kb.get(), aState, () -> getEditorCas
-                            (aHandler), aTypedString));
+    private List<KBHandle> getInstances(ConceptFeatureTraits traits, Project project,
+        AnnotatorState aState, AnnotationActionHandler aHandler, String aTypedString)
+    {
+        List<KBHandle> handles = new ArrayList<>();
+        if (traits.getRepositoryId() != null) {
+            // If a specific KB is selected, get its instances
+            Optional<KnowledgeBase> kb = kbService
+                .getKnowledgeBaseById(project, traits.getRepositoryId());
+            if (kb.isPresent()) {
+                //TODO: (#122) distinguish between concepts & instances when concept linking enabled
+                if (kb.get().isSupportConceptLinking()) {
+                    handles.addAll(
+                        listLinkingInstances(kb.get(), aState, () -> getEditorCas(aHandler),
+                            aTypedString));
+                }
+                else {
+                    if (traits.getScope() != null) {
+                        handles = kbService
+                            .listChildConceptsInstances(kb.get(), traits.getScope(), false, 50)
+                            .stream().filter(inst -> inst.getUiLabel().contains(aTypedString))
+                            .collect(Collectors.toList());
                     }
                     else {
-                        if (traits.getScope() != null) {
-                            handles = kbService.listInstances(kb.get(), traits.getScope(), false)
-                                    .stream()
-                                    .filter(inst -> inst.getUiLabel().contains(aTypedString))
-                                    .collect(Collectors.toList());
-                        }
-                        else {
-                            for (KBHandle concept : kbService.listConcepts(kb.get(), false)) {
-                                handles.addAll(kbService.listInstances(kb.get(), 
-                                    concept.getIdentifier(), false));
-                            }
+                        for (KBHandle concept : kbService.listConcepts(kb.get(), false)) {
+                            handles.addAll(
+                                kbService.listInstances(kb.get(), concept.getIdentifier(), false));
```

Comment:
This will not return a sorted list and also it doesn't take into consideration the input string: aTypedString for auto complete field.